rholang/src/main/scala/coop/rchain/rholang/interpreter/matcher/SpatialMatcher.scala

```diff
@@ -30,22 +29,29 @@ import scala.collection.mutable
 // In a few places we use StateT[Stream, FreeMap, A] for backtracking. In order
 // to help cut down on backtracking, wherever one of several possible matches
 // will do, we just take one.
-trait SpatialMatcher[T, P] {
-  def spatialMatch(target: T, pattern: P): NonDetFreeMapWithCost[Unit]
+trait SpatialMatcher[T, P, F[_]] {
```

Comment:
`F[_]` should move to the first position as it is done in all other places. For consistence